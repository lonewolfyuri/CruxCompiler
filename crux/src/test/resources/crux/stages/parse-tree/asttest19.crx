// scoped usage
var a: int;

func main() : void {
  if (false) {
    if (true) {
      while (true) {
        let a = 7;
      }
    }
  }
}
