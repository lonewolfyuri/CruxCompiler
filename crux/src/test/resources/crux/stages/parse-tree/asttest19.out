program
  declarationList
    declaration
      variableDeclaration
        type
    declaration
      functionDefinition
        parameterList
        type
        statementBlock
          statementList
            statement
              ifStatement
                expression0
                  expression1
                    expression2
                      expression3
                        expression0
                          expression1
                            expression2
                              expression3
                                literal
                statementBlock
                  statementList
                    statement
                      ifStatement
                        expression0
                          expression1
                            expression2
                              expression3
                                expression0
                                  expression1
                                    expression2
                                      expression3
                                        literal
                        statementBlock
                          statementList
                            statement
                              whileStatement
                                expression0
                                  expression1
                                    expression2
                                      expression3
                                        expression0
                                          expression1
                                            expression2
                                              expression3
                                                literal
                                statementBlock
                                  statementList
                                    statement
                                      assignmentStatement
                                        designator
                                        expression0
                                          expression1
                                            expression2
                                              expression3
                                                literal
